pkg/e2e/infra/utils.go

```diff
@@ -157,6 +157,18 @@ func getMachineList(client runtimeclient.Client) (*mapiv1beta1.MachineList, erro
 	return &machineList, nil
 }
 
+// getMachines returns the list of machines or an error
+func getMachines(client runtimeclient.Client) ([]mapiv1beta1.Machine, error) {
+	machineList, err := getMachineList(client)
+	if err != nil {
+		return nil, fmt.Errorf("error getting machineList: %v", err)
+	}
+	if len(machineList.Items) == 0 {
```

Comment:
On reflection, this doesn't seem right. There's no actual error here. There's just no machines. And that's perfectly legitimate.

Reply:
As per https://github.com/openshift/cluster-api-actuator-pkg/pull/44/files#r263715365, it's better to have consumer of `getMachines` to decide if this is an error or not.